lib/db-utils.js

```diff
@@ -187,7 +187,7 @@ module.exports = (srf, logger) => {
             gateway: selected,
             service_provider_sid: a[0].service_provider_sid,
             account_sid: a[0].account_sid,
-            application_sid: r.length ? r[0].application_sid : selected.application_sid,
+            application_sid: r.length ? r[0]?.application_sid : selected.application_sid,
```

Comment:
no, this looks wrong.  If we **do** have a row, but the application_sid is null, this would assign null rather than the carrier level application_sid